src/factory.js

```diff
@@ -100,59 +99,35 @@ export default function plotComponentFactory(Plotly) {
         });
     }
 
-    UNSAFE_componentWillUpdate(nextProps) {
+    componentDidMount() {
+      this.unmounting = false;
+
+      this.updatePlotly(true, this.props.onInitialized, true);
+    }
+
+    componentDidUpdate(prevProps) {
```

Comment:
why the rename of `nextProps` here?

Reply:
Because they are not next props, they are previous props. The function is called after the update, not before like `componentWillUpdate`.

Reply:
Basically because React API